### ISSUE TITLE
`formatNumber` returns NaN on Zero as input

### ISSUE THREAD

**Issue by steven-r** [bitburner-official/bitburner-src#956]
- Release: Bitburner 2.5.1 (05875e)

The function `ns.formatNumber` doesn't return a proper result for `n = 0`, `fractionalDigits = 0` and `minDecimalPlaces = 0`. The expected result would be `0`, but the response is `NaNundefined ` .

Example-Code:
```js
ns.tprint("Format 1:       ", ns.formatNumber(1, 0, 0))
ns.tprint("Format 0:       ", ns.formatNumber(0, 0, 0))
ns.tprint("Format 0 (def): ", ns.formatNumber(0))
```

Expected output:
```
test_fm.js: Format 1:       1
test_fm.js: Format 0:       0
test_fm.js: Format 0 (def): 0.000
```

Current Output:
```
test_fm.js: Format 1:       1
test_fm.js: Format 0:       NaNundefined
test_fm.js: Format 0 (def): 0.000
```

**Comment by steven-r** [comment]
As mentioned by TheMainMan (https://discord.com/channels/415207508303544321/415213413745164318/1180514580749885450):

looks like its due to this piece of the number formatter code

```js
if (nAbs < suffixStart) {
    if (isInteger) return basicFormatter.format(n);
    return getFormatter(fractionalDigits).format(n);
  }
```
issue goes away if I change it to 

```js
if (nAbs <= suffixStart) {
    if (isInteger) return basicFormatter.format(n);
    return getFormatter(fractionalDigits).format(n);
  }
```

New output:

```
Running script with 1 thread(s), pid 1 and args: [].
1.js: Format 1:       1
1.js: Format 0:       0
```

**Comment by LJNeon** [comment]
This has been fixed and can be closed (see #957).
